src/Components/LoginForm.tsx

```diff
@@ -0,0 +1,233 @@
+import React, {
+  FormEvent,
+  useContext,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { AppContext } from '../AppContext';
+import { findUser, addUsers } from '../api/todos';
+import { Notification } from './Notification';
+
+const addPrevUser = (prevEmail: string) => {
+  const getPrevUsers: string[] | 0 = JSON
+    .parse(localStorage.getItem('prevusers') || '0');
```

Comment:
Try to create enums or constants for all string values. It's really easy to make a mistake in these situations